tests/Transports.Subscriptions.WebSockets.Tests/WebSocketsConnectionFacts.cs

```diff
@@ -46,14 +46,8 @@ public async Task Should_accept_websocket_connection()
     [Fact]
     public async Task Should_not_accept_websocket_with_wrong_protocol()
     {
-        /* Given */
-        /* When */
-        var socket = await ConnectAsync("do-not-accept");
-        var segment = new ArraySegment<byte>(new byte[1024]);
-        var received = await socket.ReceiveAsync(segment, CancellationToken.None);
-
-        /* Then */
-        received.CloseStatus.ShouldBe(WebSocketCloseStatus.ProtocolError);
+        (await Should.ThrowAsync<InvalidOperationException>(async () => await ConnectAsync("do-not-accept")))
+            .Message.ShouldBe("Incomplete handshake, status code: 400");
```

Comment:
Previously the code would accept a websocket connection for an invalid protocol, and immediately close it with an error message.  Now it is refused at the HTTP level, returning a 400 bad result status code (with a proper message, which unfortunately is not visible through the C# WebSocket client class).